agent/agent.go

```diff
@@ -165,73 +155,86 @@ func formatBytes(val uint64) string {
 	return fmt.Sprintf("%d bytes", val)
 }
 
-func handle(conn io.Writer, msg []byte) error {
-	switch msg[0] {
+// Agent implement http.Handler
+type Agent struct{}
+
+// ServeHTTP parse the requests body into signal.Command and run the task
+func (a *Agent) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	cmd := signal.Command{}
+	decodeErr := json.NewDecoder(r.Body).Decode(&cmd)
```

Comment:
I was thinking of starting a new HTTP server at a random port (or at Options.Addr) rather than providing this custom handler. Then, we don't expect the users to start an HTTP server in order to use the agent.